### ISSUE TITLE
Span<T>(Void*, Int32) constructor Length arg is not number of bytes

### ISSUE THREAD

**Issue by leafi** [dotnet/dotnet-api-docs#2045]
https://docs.microsoft.com/en-us/dotnet/api/system.span-1.-ctor?view=netcore-2.2#System_Span_1__ctor_System_Void__System_Int32_

In this page, the length Int32 argument is described as:
`The number of bytes to be included in the Span<T>.`

As far as I can verify, this is simply not true. Execute the following code in an unsafe context in .NET Core 2.2:

```
Span<long> xx = stackalloc long[20000];
Console.WriteLine(xx.Length);
xx.Clear();
Console.WriteLine("OK");

Span<long> yy = new Span<long>(
    Marshal.AllocHGlobal(20000*sizeof(long)).ToPointer(),
    20000*sizeof(long)
);
Console.WriteLine(yy.Length);
yy.Clear();
Console.WriteLine("OK");
```

You get the output:
```
20000
OK
160000
```

The second Span, using the `Span<T>(Void*, Int32)` constructor has resulted in .Length being 8 times as big as the stackalloc'd version, and the application crashes when trying to clear the contents of the Span.

Instead, the Length arg refers to the number of elements of type T in the span.

**Comment by yatli** [comment]
either the doc or the implementation is bad.

``` C#
unsafe void Main()
{
	byte[] buf = new byte[100000000];
	fixed(byte* p = buf)
	{
		Span<double> span = new Span<double>(p, 100000000);
		var arr = span.ToArray();
		Console.WriteLine(span.ToArray().Length);
		Console.WriteLine(span[100000000 - 1]);
	}
}
```
100000000 bytes are not 100000000 doubles.

**Comment by gfoidl** [comment]
@leafi the documentation was updated in #2272, so this issue can be closed?

**Comment by rpetrusha** [comment]
This issue is a duplicate of #2271 and was fixed in #2272. Closing as duplicate.
